### PR TITLE
PT-5954: Can't extend Orders connection handler in XAPI

### PR DESCRIPTION
**Pull request by BelyakovSergey**
## Description
Can't extend Orders connection handler in XAPI
### Jira-link:

https://virtocommerce.atlassian.net/browse/PT-5954

### CODE REVIEW

File: src/XPurchase/VirtoCommerce.ExperienceApiModule.XOrder/Schemas/OrderSchema.cs [hunk 1]
```diff
@@ -194,18 +174,8 @@ public void Build(ISchema schema)
...
-            {
-                Skip = skip,
-                Take = first ?? context.PageSize ?? 10,
-                Filter = context.GetArgument<string>("filter"),
-                Sort = context.GetArgument<string>("sort"),
-                CultureName = context.GetArgument<string>(nameof(Currency.CultureName).ToCamelCase()),
-                CustomerId = context.GetArgumentOrValue<string>("userId")
-            };
+            var query = AbstractTypeFactory<SearchOrderQuery>.TryCreateInstance();
+            query.Map(context);
```
Comment (Vectorfield4) [inline src/XPurchase/VirtoCommerce.ExperienceApiModule.XOrder/Schemas/OrderSchema.cs:178]: Create extension. Consider moving Line 180 inside thisextension.
```suggestion
            var query = context.ExtractQuery<SearchOrderQuery>();
```

File: src/XPurchase/VirtoCommerce.ExperienceApiModule.XOrder/Schemas/QueryConnectionArguments.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        public QueryConnectionArguments()
+        {
+            Argument<StringGraphType>("filter", "This parameter applies a filter to the query results");
+            Argument<StringGraphType>("sort", "The sort expression");
+            Argument<StringGraphType>("cultureName", "Culture name (\"en-US\")");
+            Argument<StringGraphType>("userId", "");
+        }
+
+        public QueryConnectionArguments AddArguments(QueryArguments arguments)
```
Comment (Vectorfield4) [inline src/XPurchase/VirtoCommerce.ExperienceApiModule.XOrder/Schemas/QueryConnectionArguments.cs:15]:
```suggestion
        public virtual QueryConnectionArguments AddArguments(QueryArguments arguments)
```

File: src/XPurchase/VirtoCommerce.ExperienceApiModule.XOrder/Schemas/OrderSchema.cs [hunk 3]
```diff
@@ -78,25 +70,13 @@ public void Build(ISchema schema)
...
-            var orderConnectionBuilder = GraphTypeExtenstionHelper.CreateConnection<CustomerOrderType, object>()
-                .Name("orders")
-                .Argument<StringGraphType>("filter", "This parameter applies a filter to the query results")
-                .Argument<StringGraphType>("sort", "The sort expression")
-                .Argument<StringGraphType>("cultureName", "Culture name (\"en-US\")")
-                .Argument<StringGraphType>("userId", "")
-                .PageSize(20);
-
+            var orderConnectionBuilder = GraphTypeExtenstionHelper.CreateConnection<CustomerOrderType, object>().Name("orders").PageSize(20);
+            orderConnectionBuilder.FieldType.Arguments = AbstractTypeFactory<QueryConnectionArguments>.TryCreateInstance().AddArguments(orderConnectionBuilder.FieldType.Arguments);
```
Comment (Vectorfield4) [inline src/XPurchase/VirtoCommerce.ExperienceApiModule.XOrder/Schemas/OrderSchema.cs:74]: Hide second line with an extension. You have all data to make it.
```suggestion
            var orderConnectionBuilder = GraphTypeExtenstionHelper
                .CreateConnection<CustomerOrderType, object>()
                .Name("orders")
                .PageSize(20)
                .Arguments<QueryConnectionArguments>();
```

**Review by Vectorfield4 — APPROVED** [review]

File: src/XPurchase/VirtoCommerce.ExperienceApiModule.XOrder/Schemas/QueryConnectionArguments.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+using GraphQL.Types;
+
+namespace VirtoCommerce.ExperienceApiModule.XOrder.Schemas
+{
+    public class QueryConnectionArguments : ArgumentList
```
Comment (ksavosteev) [inline src/XPurchase/VirtoCommerce.ExperienceApiModule.XOrder/Schemas/QueryConnectionArguments.cs:5]: rename to OrderQueryConnectionArguments

File: src/VirtoCommerce.ExperienceApiModule.Core/Infrastructure/ISearchOrderQuery.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+using GraphQL.Builders;
+
+namespace VirtoCommerce.ExperienceApiModule.Core.Infrastructure
+{
+    public interface ISearchOrderQuery
```
Comment (ksavosteev) [inline src/VirtoCommerce.ExperienceApiModule.Core/Infrastructure/ISearchOrderQuery.cs:5]: rename to IExtendableQuery
move to Core

File: src/VirtoCommerce.ExperienceApiModule.Core/Infrastructure/ISearchOrderQuery.cs [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+    {
+        public void Map(IResolveConnectionContext<object> context);
```
Comment (ksavosteev) [inline src/VirtoCommerce.ExperienceApiModule.Core/Infrastructure/ISearchOrderQuery.cs:7]: use IResolveFieldContext

**Review by ksavosteev — CHANGES_REQUESTED** [review]

**Review by ksavosteev — APPROVED** [review]

**Pull request merged**